components/Buttons/src/MDCFloatingButton+Animation.m

```diff
@@ -89,110 +86,190 @@ + (float)fab_dragCoefficient {
 }
 #endif
 
++ (CFTimeInterval)fab_animationDurationFromWallTimeDuration:(CFTimeInterval)duration
+                                                   forLayer:(CALayer *)layer {
+  CFTimeInterval finalDuration = duration;
+
+#if TARGET_IPHONE_SIMULATOR
+  finalDuration *= [self fab_dragCoefficient];
+#endif
+
+  return [layer convertTime:finalDuration fromLayer:nil];
+}
+
++ (CFTimeInterval)fab_animationBeginTimeFromWallTimeOffset:(CFTimeInterval)offset
+                                                  forLayer:(CALayer *)layer {
+  CFTimeInterval beginTime = offset;
+
+#if TARGET_IPHONE_SIMULATOR
+  beginTime *= [self fab_dragCoefficient];
+#endif
+
+  beginTime += CACurrentMediaTime();
+  return [layer convertTime:beginTime fromLayer:nil];
+}
+
 - (void)expand:(BOOL)animated completion:(void (^_Nullable)(void))completion {
+  // If both X- and Y-scale values are 1 or greater, do not expand
+  if (self.layer.transform.m11 >= 1 && self.layer.transform.m22 >= 1) {
+    NSAssert(NO, @"The FAB should not be expanded unless its X and Y scales are less than 1.");
+    return;
+  }
```

Comment:
Could this be enforced by keep track of an expanded state? Or are you concerned that others may apply their own expansion-like animations to the view?

Reply:
I thought it was a bit weird to add a property to MDCFloatingButton that was only being used in this category.  I can be persuaded to do that if we have an established precedent.